### PR TITLE
Feature/results analysis rebased

### PR DESCRIPTION
**Pull request by tleyden**
Rebased version of https://github.com/couchbaselabs/mobile-testkit/pull/680

### CODE REVIEW

**Comment by tleyden** [comment]
I just pushed a few commits which address the pylint errors.  

`validate_code.sh` output:

```
$ ./validate_code.sh
pylint -E keywords/*.py
No config file found, using default configuration
pylint -E libraries/*.py
No config file found, using default configuration
pylint -E libraries/data/*.py
No config file found, using default configuration
pylint -E libraries/provision/*.py
No config file found, using default configuration
************* Module libraries.provision.ec2
E:439,11: Instance of 'Ec2Inventory' has no 'group_by_instance_id' member (no-member)
E:445,11: Instance of 'Ec2Inventory' has no 'group_by_region' member (no-member)
E:451,11: Instance of 'Ec2Inventory' has no 'group_by_availability_zone' member (no-member)
E:454,19: Instance of 'Ec2Inventory' has no 'group_by_region' member (no-member)
E:459,11: Instance of 'Ec2Inventory' has no 'group_by_ami_id' member (no-member)
E:466,11: Instance of 'Ec2Inventory' has no 'group_by_instance_type' member (no-member)
E:473,11: Instance of 'Ec2Inventory' has no 'group_by_key_pair' member (no-member)
E:480,11: Instance of 'Ec2Inventory' has no 'group_by_vpc_id' member (no-member)
E:487,11: Instance of 'Ec2Inventory' has no 'group_by_security_group' member (no-member)
E:499,11: Instance of 'Ec2Inventory' has no 'group_by_tag_keys' member (no-member)
E:508,36: Instance of 'Ec2Inventory' has no 'group_by_route53_names' member (no-member)
E:516,11: Instance of 'Ec2Inventory' has no 'group_by_tag_none' member (no-member)
E:546,11: Instance of 'Ec2Inventory' has no 'group_by_instance_id' member (no-member)
E:552,11: Instance of 'Ec2Inventory' has no 'group_by_region' member (no-member)
E:558,11: Instance of 'Ec2Inventory' has no 'group_by_availability_zone' member (no-member)
E:561,19: Instance of 'Ec2Inventory' has no 'group_by_region' member (no-member)
E:566,11: Instance of 'Ec2Inventory' has no 'group_by_instance_type' member (no-member)
E:573,11: Instance of 'Ec2Inventory' has no 'group_by_vpc_id' member (no-member)
E:580,11: Instance of 'Ec2Inventory' has no 'group_by_security_group' member (no-member)
E:594,11: Instance of 'Ec2Inventory' has no 'group_by_rds_engine' member (no-member)
E:600,11: Instance of 'Ec2Inventory' has no 'group_by_rds_parameter_group' member (no-member)
E:711,15: Instance of 'list' has no 'host' member (no-member)
E:714,19: Instance of 'list' has no 'host' member (no-member)
E:718,43: Instance of 'list' has no 'host' member (no-member)
pylint -E libraries/testkit/*.py
No config file found, using default configuration
pylint -E libraries/utilities/*.py
No config file found, using default configuration
************* Module libraries.utilities.generate_clusters_from_pool
E:160,27: Module 'netifaces' has no 'ifaddresses' member (no-member)
E:162,27: Module 'netifaces' has no 'ifaddresses' member (no-member)
************* Module libraries.utilities.monkeyrunner
E:  8, 0: Unable to import 'com.android.monkeyrunner' (import-error)
pylint -E testsuites/*.py
No config file found, using default configuration
pylint -E testsuites/android/listener/*.py
No config file found, using default configuration
pylint -E testsuites/listener/shared/client_client/*.py
No config file found, using default configuration
pylint -E testsuites/listener/shared/client_sg/*.py
No config file found, using default configuration
pylint -E testsuites/syncgateway/*.py
No config file found, using default configuration
pylint -E testsuites/syncgateway/functional/shared/*.py
No config file found, using default configuration
pylint -E testsuites/syncgateway/performance/*.py
No config file found, using default configuration
pylint -E testsuites/syncgateway/functional/**/*.py
No config file found, using default configuration
```

**Comment by tleyden** [comment]
Successful perf run: http://uberjenkins.sc.couchbase.com/view/Performance/job/sync-gateway-perf-test/43/

**Comment by tleyden** [comment]
Functional test run (in progress): http://uberjenkins.sc.couchbase.com/view/QE%20Dev/job/syncgateway-functional-tests-dev/184/console

File: libraries/provision/ansible/playbooks/start-sgload.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  become: yes
+  vars:
+    sync_gateway_node: "{{ hostvars[groups.sync_gateways[0]].ansible_host }}"
+
+  # Deploy sgload script and start it
+  tasks:
+    - debug: msg={{ sync_gateway_node }}
+    - name: copy the run sgload script
+      copy: src=files/run_sgload.sh dest=/home/centos/ owner=centos group=centos mode=0755
+    - name: start sgload in screen
```
Comment (sethrosetter) [inline libraries/provision/ansible/playbooks/start-sgload.yml]: I thought we were trying to avoid using screen?

Comment (tleyden) [inline libraries/provision/ansible/playbooks/start-sgload.yml]: Fixed in https://github.com/couchbaselabs/mobile-testkit/pull/743/commits/05065bbb825d33b36bbbb016e379e7a1e568f7aa -- was dead code

File: testsuites/syncgateway/performance/run_sgload_perf_test.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    if not known_args.skip_build_sgload:
+        build_sgload(main_ansible_runner)
+
+    # get load generator and sg hostnames
+    lg_hosts_main = get_load_generators_hosts(main_cluster_config)
+    sg_hosts_main = get_sync_gateways_hosts(main_cluster_config)
+
+    # Get a map from load generator hostnames to sync gateway hostnames
+    # eg, {'192.168.33.13': '192.168.33.11'} key=lg, val=sg
+    lg2sg = get_lg2sg_map(lg_hosts_main, sg_hosts_main)
```
Comment (tleyden) [inline testsuites/syncgateway/performance/run_sgload_perf_test.py]: Fixed in https://github.com/couchbaselabs/mobile-testkit/pull/743/commits/b1570557525cb9c47ea61ce19477cf1760d4cb08

**Comment by tleyden** [comment]
Re-validating via:

* http://uberjenkins.sc.couchbase.com/view/QE%20Dev/job/syncgateway-functional-tests-dev/186/console
* http://uberjenkins.sc.couchbase.com/view/Performance/job/sync-gateway-perf-test/47/console

**Comment by tleyden** [comment]
Ok both functional tests and perf tests have passed on latest commit.  For passing perf test, see http://uberjenkins.sc.couchbase.com/view/Performance/job/sync-gateway-perf-test/49/console

File: testsuites/syncgateway/performance/run_sgload_perf_test.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    # build_sgload (ansible)
+    if not known_args.skip_build_sgload:
+        build_sgload(main_ansible_runner)
+
+    # get load generator and sg hostnames
+    lg_hosts_main = get_load_generators_hosts(main_cluster_config)
+    sg_hosts_main = get_sync_gateways_hosts(main_cluster_config)
+
+    # Get a map from load generator hostnames to sync gateway hostnames
+    # eg, {'192.168.33.13': '192.168.33.11'} key=lg, val=sg
```
Comment (sethrosetter) [inline testsuites/syncgateway/performance/run_sgload_perf_test.py]: Is this comment still relevant?

**Pull request merged**